code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -185,12 +184,10 @@
 		var/mob/living/carbon/human/H = target
 		if(!H.vision_type)
 			to_chat(H, "<span class='notice'>You shift the nerves in your eyes, allowing you to see in the dark.</span>")
-			H.vision_type = new vision_path
-			H.update_sight()
+			H.set_sight(new /datum/vision_override/nightvision)
 		else
 			to_chat(H, "<span class='notice'>You return your vision to normal.</span>")
-			H.vision_type = null
-			H.update_sight()
+			H.set_sight()
```

Comment:
Will be best to add `null` here as param. The effect is the same but this way it makes it clear to the reader that you mean to set it to `null`.